### PR TITLE
Adding HepMC 3

### PR DESCRIPTION
**Pull request by henryiii**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@chrisburr , @gartung , I've started this formula. It would be nice if we could turn on ROOTIO, but adding the entire ROOT set of dependencies is probably overkill. What about that root-base package idea? Also let me know about listing you on the recipe.

### CODE REVIEW

File: recipes/hepmc3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  summary: 'HepMC3 is a new rewrite of HepMC event record.'
+  description: |
+      HepMC3 is a new rewrite of HepMC event record. It uses shared pointers
+      for in-memory navigation and the POD concept for the persistency.
+  dev_url: https://gitlab.cern.ch/hepmc/HepMC3
+
+extra:
+  recipe-maintainers:
+    - henryiii
+    - chrisburr
```
Comment (chrisburr) [inline recipes/hepmc3/meta.yaml]: :+1:

File: recipes/hepmc3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - {{ compiler('fortran') }}
+  commands:
+    # Workaround for lack of CONDA_BUILD_SYSROOT variable in staged recipes
+    - export CONDA_BUILD_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk  # [osx]
+    - cd source/examples/BasicExamples
+    - ${CXX} basic_tree.cc -o basic_tree.exe -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib -lHepMC3
+    - ./basic_tree.exe
+
+about:
+  home: http://github.com/simplejson/simplejson
```
Comment (chrisburr) [inline recipes/hepmc3/meta.yaml]:
```suggestion
  home: https://hepmc.web.cern.ch/hepmc/
```

Comment (henryiii) [reply recipes/hepmc3/meta.yaml]: :blush:

File: recipes/hepmc3/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+# Enable bash strict mode
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+mkdir -p build
+cd build
+
+# ROOTIO is currently off, this could be added after ROOT adds a root-base formula
```
Comment (chrisburr) [inline recipes/hepmc3/build.sh]: It seems like a good idea, I should have some time to think about it in May.

File: recipes/hepmc3/meta.yaml [hunk 4]
```diff
@@ -29,11 +29,11 @@ test:
     # Workaround for lack of CONDA_BUILD_SYSROOT variable in staged recipes
     - export CONDA_BUILD_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk  # [osx]
     - cd source/examples/BasicExamples
-    - ${CXX} basic_tree.cc -o basic_tree.exe -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib -lHepMC3
+    - ${CXX} basic_tree.cc -o basic_tree.exe -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib64 -lHepMC3
```
Comment (chrisburr) [inline recipes/hepmc3/meta.yaml]: Can you configure/patch it so this is installed in `lib` instead of `lib64`?

File: recipes/hepmc3/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  description: |
+      HepMC3 is a new rewrite of HepMC event record. It uses shared pointers
+      for in-memory navigation and the POD concept for the persistency.
+  dev_url: https://gitlab.cern.ch/hepmc/HepMC3
+
+extra:
+  recipe-maintainers:
+    - henryiii
+    - chrisburr
+    - gartung
```
Comment (gartung) [inline recipes/hepmc3/meta.yaml]: I am OK with this.

**Comment by henryiii** [comment]
Mostly working, just have the following error:

```
dyld: Library not loaded: @rpath/libHepMC3.dylib
  Referenced from: /Users/vsts/agent/2.150.3/work/1/a/miniconda/conda-bld/hepmc3_1557072362994/test_tmp/source/test/./testMass.exe
  Reason: image not found
```

The line that's making the broken build is:

```bash
- ${CXX} testMass.cc -o testMass.exe -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib -lHepMC3
```

Any ideas?

**Comment by henryiii** [comment]
@chrisburr This one is ready for review now.

**Pull request merged**